src/components/search.js

```diff
@@ -16,7 +12,7 @@ const Search = () => {
           <line x1="17.1" y1="14.7" x2="22.7" y2="18.9" stroke="black" strokeWidth="3" strokeLinecap="round"/>
         </svg>
       </div>
-      <form onSubmit={handleSubmit} action="/search">
```

Comment:
that was the right choice here. 
You're fighting the component lifecycle doing this and in turn breaking the React framework mentioned above.